qa/purchase_crypto_listing.py

```diff
@@ -0,0 +1,176 @@
+        resp = json.loads(r.text)
+        if resp["state"] != "AWAITING_FULFILLMENT":
+            raise TestFailure("PurchaseCryptoListingTest - FAIL: Buyer failed to detect his payment")
+        if resp["funded"] == False:
+            raise TestFailure("PurchaseCryptoListingTest - FAIL: Buyer incorrectly saved as unfunded")
+        if resp["contract"]["vendorListings"][0]["metadata"]["coinType"] != "ETH":
+            raise TestFailure("PurchaseCryptoListingTest - FAIL: Buyer incorrectly saved without a coinType")
+        if resp["contract"]["buyerOrder"]["items"][0]["paymentAddress"] != "crypto_payment_address":
+            raise TestFailure("PurchaseCryptoListingTest - FAIL: Buyer incorrectly saved without a paymentAddress")
+
+        # check vendor detected payment
+        api_url = vendor["gateway_url"] + "ob/order/" + orderId
+        r = requests.get(api_url)
+        if r.status_code != 200:
+            raise TestFailure("PurchaseCryptoListingTest - FAIL: Couldn't load order from Vendor")
+        resp = json.loads(r.text)
+        if resp["state"] != "AWAITING_FULFILLMENT":
+            raise TestFailure("PurchaseCryptoListingTest - FAIL: Vendor failed to detect payment")
+        if resp["funded"] == False:
+            raise TestFailure("PurchaseCryptoListingTest - FAIL: Vendor incorrectly saved as unfunded")
+
+        with open('testdata/fulfillment_crypto.json') as fulfillment_file:
+            fulfillment_json = json.load(fulfillment_file, object_pairs_hook=OrderedDict)
+        fulfillment_json["orderId"] = orderId
+        fulfillment_json["slug"] = slug
+        api_url = vendor["gateway_url"] + "ob/orderfulfillment"
+        r = requests.post(api_url, data=json.dumps(fulfillment_json, indent=4))
+        if r.status_code == 404:
+            raise TestFailure("PurchaseCryptoListingTest - FAIL: Fulfillment post endpoint not found")
+        elif r.status_code != 200:
+            resp = json.loads(r.text)
+            raise TestFailure("PurchaseCryptoListingTest - FAIL: Fulfillment POST failed. Reason: %s", resp["reason"])
+
+        # check buyer received fulfillment
+        api_url = buyer["gateway_url"] + "ob/order/" + orderId
+        r = requests.get(api_url)
+        if r.status_code != 200:
+            raise TestFailure("PurchaseCryptoListingTest - FAIL: Couldn't load order from Buyer")
+        resp = json.loads(r.text)
+        if resp["state"] != "FULFILLED":
+            raise TestFailure("PurchaseCryptoListingTest - FAIL: Buyer failed to detect order fulfillment")
+        if resp["contract"]["vendorOrderFulfillment"][0]["cryptocurrencyDelivery"][0]["transactionID"] != "crypto_transaction_id":
+            raise TestFailure("PurchaseCryptoListingTest - FAIL: Buyer failed to detect transactionID")
+
+        print("PurchaseCryptoListingTest - PASS")
+
+if __name__ == '__main__':
+    print("Running PurchaseCryptoListingTest")
+    PurchaseCryptoListingTest().main(["--regtest"])
```

Comment:
I think all tests should be using this option? I'm surprised to find this here. Is there a better location which includes this option on every test instead of just this one?

Reply:
By default flags are `["--disablewallet", "--testnet", "--disableexchangerates"]`. We could add `--regtest` but it would still need to be overridden in all the places that don't want --disablewallet. We need a better way to declaring which flags a test should use.

Maybe have the same defaults but be able to do like `PurchaseCryptoListingTest().WithWallet().WithExchangeRates()`